addons/pos_sale/models/pos_order.py

```diff
@@ -38,8 +37,8 @@ def _prepare_invoice_vals(self):
             else:
                 addr = self.partner_id.address_get(['delivery'])
                 invoice_vals['partner_shipping_id'] = addr['delivery']
-            if sale_orders[0].payment_term_id:
-                invoice_vals['invoice_payment_term_id'] = False
+            if sale_orders[0].payment_term_id and not sale_orders[0].payment_term_id.early_discount:
+                invoice_vals['invoice_payment_term_id'] = sale_orders[0].payment_term_id.id
```

Comment:
If you don't specify a `false` value, the `_compute_invoice_payment_term_id` will be triggered and will get the payment term from the partner. If the payment term set on the partner is an early payment discount, you still have a problem here.
Note: This was already the case before but I guess no one create a SO without any payment term. However, since the field is not required on `sale.order`, it's better to take this case under control
```suggestion
            if sale_orders[0].payment_term_id and not sale_orders[0].payment_term_id.early_discount:
                invoice_vals['invoice_payment_term_id'] = sale_orders[0].payment_term_id.id
            else:
                invoice_vals['invoice_payment_term_id'] = False
```